apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java

```diff
@@ -100,70 +99,56 @@ public GeneratorServiceImpl(PropertiesHolder propertiesHolder,
         GeneratorMemoryEntity generator = new GeneratorMemoryEntity(keySeed, publicKey, accountId);
-        globalSync.updateLock();
-        try {
-            if (blockchain.getHeight() >= blockchainConfig.getLastKnownBlock()) {
-                this.setLastBlock(blockchain.getLastBlock(), generator);
-                if (generateBlocksTask != null) {
-                    generateBlocksTask.resetSortedForgers();
-                }
-            }
-        } finally {
-            globalSync.updateUnlock();
+        if (blockchain.getHeight() >= blockchainConfig.getLastKnownBlock()) {
+            this.setLastBlock(blockchain.getLastBlock(), generator);
+            setConditionToResetForgers();
         }
         GeneratorMemoryEntity old = generators.putIfAbsent(generator.getAccountId(), generator);
         if (old != null) {
-            log.debug(old + " is already forging");
+            log.debug("{} is already forging ({} ms)", old, (System.currentTimeMillis() - start));
             return old;
         }
-        log.debug(generator + " started");
+        log.debug("{} started ({} ms)", generator, (System.currentTimeMillis() - start));
         return generator;
     }
 
     @Override
     public GeneratorMemoryEntity stopForging(byte[] keySeed) {
-        long start = System.currentTimeMillis();
         GeneratorMemoryEntity generator = generators.remove(Convert.getId(Crypto.getPublicKey(keySeed)));
-        if (generator != null && generateBlocksTask != null) {
-            globalSync.updateLock();
-            try {
-                generateBlocksTask.resetSortedForgers();
-            } finally {
-                globalSync.updateUnlock();
-            }
-            log.debug(generator + " stopped ({} ms)", (System.currentTimeMillis() - start));
+        if (generator != null) {
+            log.debug(generator + ": Stop Forging command added to the queue.");
+            setConditionToResetForgers();
         }
         return generator;
     }
 
     @Override
     public int stopForging() {
-        long start = System.currentTimeMillis();
         int count = generators.size();
         Iterator<GeneratorMemoryEntity> iter = generators.values().iterator();
         while (iter.hasNext()) {
             GeneratorMemoryEntity generator = iter.next();
             iter.remove();
             log.debug(generator + " stopped");
         }
-        globalSync.updateLock();
-        try {
-            if (generateBlocksTask != null) {
-                generateBlocksTask.resetSortedForgers();
-            }
-        } finally {
-            globalSync.updateUnlock();
-        }
-        log.debug("stopped forging ({} ms)", (System.currentTimeMillis() - start));
+        log.debug("Admin: Stop Forging command added to the queue.");
+        setConditionToResetForgers();
         return count;
     }
 
+    private void setConditionToResetForgers() {
+        if (generateBlocksTask != null) {
```

Comment:
Probably the check: `!propertiesHolder.isLightClient()` could bring more clarity here

Reply:
added